packages/react/src/components/Alert/Alert.tsx

```diff
@@ -0,0 +1,134 @@
+
+  /** Icon used for dismissing the Alert, 'close' by default. */
+  closeIcon?: ShorthandValue
+
+  /** An alert may be formatted to display a danger message. */
+  danger?: boolean
+
+  /** An alert may be formatted to display information. */
+  info?: boolean
+
+  /**
+   * The callback used when the user chooses to hide the Alert.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onClose?: ComponentEventHandler<AlertProps>
+
+  /** An alert may be formatted to display a successful message. */
+  success?: boolean
+
+  /** An alert may be formatted to display a warning message. */
+  warning?: boolean
+}
+
+/**
+ * A Alert displays information that explains nearby content.
+ */
+class Alert extends UIComponent<ReactProps<AlertProps>, {}> {
+  static displayName = 'Alert'
+  static className = 'ui-alert'
+
+  static slotClassNames: AlertSlotClassNames = {
+    content: `${Alert.className}__content`,
+    closeIcon: `${Alert.className}__closeIcon`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon(),
+    attached: PropTypes.oneOfType([PropTypes.bool, PropTypes.oneOf(['top', 'bottom'])]),
+    closeIcon: customPropTypes.itemShorthand,
+    danger: PropTypes.bool,
+    info: PropTypes.bool,
+    onClose: PropTypes.func,
+    success: PropTypes.bool,
+    warning: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    accessibility: defaultBehavior,
+    closeIcon: 'close',
+  }
+
+  renderComponent(config: RenderResultConfig<AlertProps>) {
+    const { accessibility, classes, ElementType, unhandledProps } = config
+    const { children, content } = this.props
+    return (
+      <ElementType
+        className={classes.root}
+        {...accessibility.attributes.root}
+        {...rtlTextContainer.getAttributes({ forElements: [children, content] })}
+        {...unhandledProps}
+      >
+        {childrenExist(children) ? children : this.renderContent(config)}
+      </ElementType>
+    )
+  }
+
+  private renderContent = ({ styles, variables }: RenderResultConfig<AlertProps>) => {
+    const { closeIcon, content, onClose } = this.props
+    return (
+      <>
+        {Box.create(content, {
+          defaultProps: {
+            className: Alert.slotClassNames.content,
+            styles: styles.content,
+          },
+        })}
+        {onClose &&
+          Icon.create(closeIcon, {
+            defaultProps: {
+              className: Alert.slotClassNames.closeIcon,
+              styles: styles.icon,
+              variables: variables.icon,
+            },
+            overrideProps: this.handleCloseIconOverrides,
+          })}
+      </>
+    )
+  }
+
+  private handleCloseIconOverrides = (predefinedProps: AlertProps) => ({
+    onClick: (e: React.SyntheticEvent) => {
+      this.handleOnClose(e)
+      _.invoke(predefinedProps, 'onClick', e, this.props)
+    },
+    ...(predefinedProps.onClick && { tabIndex: '0' }),
```

Comment:
Oops, should be this part of accessibility behavior? Can we add todo in this case?

Reply:
I removed it as we need to use `Button`, but I am not sure now how to handle the `closeIcon` prop; I think it needs to become `closeButton`; what do you think?

Reply:
If it represents `Button`, I'm fine to have `closeButton`

Reply:
changed to `closeButton`